main.cpp

```diff
@@ -18,13 +18,23 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+    Rewriter& _rewriter;
+
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& rewriter) : _rewriter(rewriter) {
         // Your code goes here
     };
 
-    void run(const MatchFinder::MatchResult &Result) override {
+    void run(const MatchFinder::MatchResult& Result) override {
         // Your code goes here
+        if (auto castExpr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast")) {
+            auto lParen = castExpr->getLParenLoc();
+            auto rParen = castExpr->getRParenLoc();

```

Comment:
Please describe the difference between the use of `getRParenLoc()` and `getSubExprAsWritten()->getBeginLoc()`.

Reply:
In this case:
getRParenLoc() gets the location of the closing brace before the variable.
getSubExprAsWritten()->getBeginLoc() gets the starting location of the variable, there may be spaces between the closing brace and the variable.